integrations/gitea.js

```diff
@@ -0,0 +1,179 @@
+const util = require('util');
+const fetch = require('node-fetch');
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+module.exports = async function(robot, kredits) {
+
+  function messageRoom(message) {
+    robot.messageRoom(process.env.KREDITS_ROOM, message);
+  }
+
+  robot.logger.debug('[hubot-kredits] Loading Gitea integration...');
+
+  let repoBlackList = [];
+  if (process.env.KREDITS_GITEA_REPO_BLACKLIST) {
+    repoBlackList = process.env.KREDITS_GITEA_REPO_BLACKLIST.split(',');
+    robot.logger.debug('[hubot-kredits] Ignoring Gitea actions from ', util.inspect(repoBlackList));
+  }
+
+  const Contributor = kredits.Contributor;
+  const Contribution = kredits.Contribution;
+
+  function getContributorByGiteaUser(username) {
+    return Contributor.all().then(contributors => {
+      const contrib = contributors.find(c => {
+        return c.gitea_username === username;
+      });
```

Comment:
there are contract methods for that: https://github.com/67P/kredits-contracts/blob/master/lib/contracts/contributor.js#L20-L38
should be something like: `Contributor.findByAccount({gitea_username: username, site: 'gitea something})`

Reply:
This is copied from the GitHub integration code, so I won't refactor things for this functionality now. Can do that after we're not losing Gitea kredits anymore.

Reply:
ok, maybe we should make an issue to not forget it. also that we don't use `gitea_username` here but the `accouts` array/hash somehow.